torch/nn/parallel/distributed.py

```diff
@@ -690,6 +696,41 @@ def __init__(
         if static_graph:
             self._set_static_graph()
 
+        # Check if user has used apply_optim_in_backward to overlap optimizer
+        # step + DDP backward. Currnent constraints:
+        # 1. Only allreduce is supported at the moment, no custom communication.
+        # 2. The reducer by default sets all grads for parameters DDP manages to
+        # None after they have been applied by the optimizer. There is no support
+        # for setting only some parameter grads to None, this must be done manually
+        # by user (and DDP_OVERLAPPED_OPTIM_SET_GRADS_TO_NONE=0 needs to be set.)
+        if any(
+            hasattr(p, '_in_backward_optimizers') for p in self.module.parameters()
```

Comment:
Should we filter ignored parameters instead of directly using `self.module.parameters()`?

Reply:
yes, great catch!